### DISCUSSION TITLE
How to reply to a post (via API)?

### DISCUSSION THREAD

**Discussion by GreenFootballs** [bluesky-social/atproto#1180] [Dev Debugging] [answered]
I know I'm probably missing something obvious, but how exactly would I use the API to reply to a specific post? I assume I'd need to refer to the post's CID in a `createRecord` call, but how?

**Comment by snarfed** [comment, marked as answer]
I believe you'd include `app.bsky.feed.post#replyRef` in a `app.bsky.feed.post`, filling in `parent` with the post you're replying to and `root` to be the same as that post's root. Lexicon:

https://github.com/bluesky-social/atproto/blob/bf4ac79cef9ce36113ac9daae65c1315983ee263/lexicons/app/bsky/feed/post.json#L36-L43

More:

* example JSON: https://github.com/snarfed/granary/blob/main/granary/tests/testdata/comment_inreplyto_id.bsky.json#L7-L23
* example Python code using @MarshalX's atproto lib: https://github.com/MarshalX/atproto/blob/main/examples/send_reply.py

**Reply by dholms** [reply]
yup exactly!

& just to add on, if those post that you're replying to is not a reply itself, then both `parent` & `root` will point to the same post

**Reply by bnewbold** [reply]
FWIW, there is a longer worked example of "how to reply" in this blog post: https://atproto.com/blog/create-post
